src/kernel-install/60-ukify.install.in

```diff
@@ -171,6 +171,17 @@ def kernel_cmdline(opts) -> str:
     return ' ' + ' '.join(options)
 
 
+def initrd_list(opts) -> list[Path]:
+    initrds = opts.initrd
+    microcode = list(opts.staging_area.glob('microcode/*'))
+    initrds = microcode + initrds
+
+    initrd = opts.staging_area / 'initrd'
+    if initrd.is_file():
+        initrds.append(initrd)
+    return initrds
```

Comment:
Hmm, `90-loaderentry.install` uses `"${KERNEL_INSTALL_STAGING_AREA}"/microcode* "${@}" "${KERNEL_INSTALL_STAGING_AREA}"/initrd*`, while your patch only looks for `opts.staging_area / 'initrd'`. I think it should implement the same logic as `90-loaderentry.install`.

And the glob needs to be sorted. Not sorting is an error in `90-loaderentry.install` that needs to be fixed.

Also, instead of building the array in steps, maybe use the following pattern:
```python
microcode = sorted(opts.staging_area.glob('microcode/*'))
initrd = sorted(opts.staging_area.glob('initrd*'))

return [*microcode, *opts.initrd, *initrd]
```

Reply:
> And the glob needs to be sorted. Not sorting is an error in `90-loaderentry.install` that needs to be fixed.

It seems the shell glob sorts the output. So `90-loaderentry.install` is OK. But here we need to sort, because the python glob seems to return files in filesystem order, not sorted at all.

Reply:
I did not notice the glob in initrd in loader. I have made the adjustment based on your suggestions. I have amended the commit with the changes